### PR TITLE
QAE-287 - E2E test case for off canvas menu hide on tablet and mobile option

### PR DESCRIPTION
**Pull request by SimranChougule1**
### Description

Added test case for off canvas menu hide on tablet and mobile options

### Screenshots

Hide on tablet -                                                       
Disable: https://share.getcloudapp.com/eDulxpLw
Enable: https://share.getcloudapp.com/wbuYmdep

Hide on mobile -
Disable: https://share.getcloudapp.com/NQuxZmvq
Enable: https://share.getcloudapp.com/v1uLrzmb 

### How has this been tested?

npm run test:e2e:interactive -- tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js

### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { createURL, createNewPost, publishPost } from '@wordpress/e2e-test-utils';
+import { setCustomize } from '../../../../../../utils/customize';
+import { setBrowserViewport } from '../../../../../../utils/set-browser-viewport';
+describe( 'off canvas menu design settings in the customizer', () => {
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js:4]: describe block's 1st letter should be capital

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	it( 'off canvas hide on tablet and mobile should apply corectly for after header', async () => {
+		const Offcanvas = {
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js:6]: constant name should be test case specific

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { createURL, createNewPost, publishPost } from '@wordpress/e2e-test-utils';
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js:1]: check the util created for "publish-post"

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js (hunk 2 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js]: Done

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js (hunk 3 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js]: Done

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js (hunk 1 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js]: Done

**Review by bsfsnehal — APPROVED** [review]

File: tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+			await createNewPost( { postType: 'post', title: 'test-1' } );
+			await createNewPost( { postType: 'page', title: 'test-2' } );
+			ppStatus = await publishPost();
+		}
+		await page.goto( createURL( '/' ), {
+			waitUntil: 'networkidle0',
+		} );
+		await setBrowserViewport( 'medium' );
+		await page.click( '.main-header-menu-toggle' );
+		await page.waitForSelector( '.ast-builder-menu-mobile .main-navigation' );
```
Comment (GauravKhupse) [inline tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js:23]: Can we please check **display :block** for show and **display :none** for hiding changes, preview not showing output.
selector:- https://share.bsf.io/wbuzPAwj

Comment (bsfsnehal) [reply tests/e2e/specs/customizer/header-builder/header-types/mobile-header/off-canvas-menu/hide-on-tablet-mobile.test.js:23]: As perviously discussed with team, the correct way is using of "tobeNull()

**Review by GauravKhupse — APPROVED** [review]

**Pull request merged**